lodash.js

```diff
@@ -2568,17 +2568,21 @@
      * @param {boolean[]} orders The sort orders of `props`.
      * @returns {Array} Returns the new sorted array.
      */
-    function baseSortByOrder(collection, props, orders) {
+    function baseSortByOrder(collection, comparitors, orders) {
       var index = -1,
           length = collection.length,
           result = isLength(length) ? Array(length) : [];
 
+      comparitors = arrayMap(comparitors, function(comparitor) {
+        return getCallback(comparitor, null, 1);
+      });
+
       baseEach(collection, function(value) {
         var length = props.length,
             criteria = Array(length);
 
         while (length--) {
-          criteria[length] = value == null ? undefined : value[props[length]];
+          criteria[length] = comparitors[length](value);
         }
```

Comment:
Yep, that was the other thing - should context param be added to
sorybyorder and sortByAll?
On Apr 12, 2015 3:48 AM, "John-David Dalton" <notifications@github.com>
wrote:

> In lodash.js
> <https://github.com/lodash/lodash/pull/1126#discussion_r28201496>:
>
> >        baseEach(collection, function(value) {
> >          var length = props.length,
> >              criteria = Array(length);
> >
> >          while (length--) {
> > -          criteria[length] = value == null ? undefined : value[props[length]];
> > +          criteria[length] = comparitors[length](value);
> >          }
>
> The one thing that bothers me is there's no this binding. It wouldn't be
> possible. This makes it a bit awkward compared to _.sortBy.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/lodash/lodash/pull/1126/files#r28201496>.
>